src/qz/utils/ArgValue.java

```diff
@@ -0,0 +1,135 @@
+package qz.utils;
+
+import qz.common.Constants;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+
+import static qz.utils.ArgValue.ArgType.*;
+
+public enum ArgValue {
+    // Informational
+    HELP(INFORMATION, "--help", "-h", "Display help information and exit.", null),
+    VERSION(INFORMATION, "--version", "-v", "Display version information and exit.", null),
+    BUNDLEID(INFORMATION, "--bundleid", "-i", "Display Apple bundle identifier and exit.", null),
+    LIBINFO(INFORMATION, "--libinfo", "-l", "Display detailed library version information and exit.", null),
+
+    // Actions
+    ALLOW(ACTION,"--allow", "--whitelist", "-a", String.format("Add the specified certificate to %s.dat.", Constants.ALLOW_FILE), "--allow cert.pem"),
+    BLOCK(ACTION,"--block", "--blacklist", "-b", String.format("Add the specified certificate to %s.dat.", Constants.BLOCK_FILE), "--block cert.pem"),
+
+    // Options
+    AUTOSTART(OPTION,"--honorautostart", "-A", "Read and honor any autostart preferences before launching.", null),
+    HEADLESS(OPTION, "--headless", "Force startup \"headless\" without graphical interface or interactive components.", null),
+
+    // Installer stubs
+    PREINSTALL(INSTALLER, "preinstall", "Perform critical pre-installation steps: Stop instances, all other special considerations.", null),
+    INSTALL(INSTALLER, "install", "Copy to the specified destination and preforms platform-specific registration.", "install --dest /my/install/location [--silent]"),
+    CERTGEN(INSTALLER, "certgen", "Performs certificate generation and registration for proper HTTPS support.", "certgen [--key key.pem --cert cert.pem] [--pfx cert.pfx --pass 12345] [--host \"list;of;hosts\"]"),
+    UNINSTALL(INSTALLER, "uninstall", "Perform all uninstall tasks: Stop instances, delete files, unregister settings.", null),
+    SPAWN(INSTALLER, "spawn", "Spawn an instance of the specified program as the logged-in user, avoiding starting as the root user if possible.", "spawn [program params ...]");
+
+    private ArgType argType;
+    private String[] matches;
+    private String description;
+    private String usage;
+
+    ArgValue(ArgType argType, String[] matches, String description, String usage) {
+        this.argType = argType;
+        this.matches = matches;
+        this.description = description;
+        this.usage = usage;
+    }
+    ArgValue(ArgType argType, String match1, String match2, String match3, String description, String usage) {
+        this(argType, new String[] { match1, match2, match3 }, description, usage);
+    }
+    ArgValue(ArgType argType, String match1, String match2, String description, String usage) {
+        this(argType, new String[] { match1, match2 }, description, usage);
+    }
+    ArgValue(ArgType argType, String match, String description, String usage) {
+        this(argType, new String[] { match }, description, usage);
+    }
```

Comment:
Why not `ArgValue(ArgType argType, String description, String usage, String... matches) {` and avoid multiple constructors like this?

Reply:
Because the compiler can't handle the difference between indeterminate strings, I'd have to inject a non-string in the middle. 😕

Reply:
I'm not sure I follow - are you saying it can't have a vararg as the last argument if the preceding argument is the same type?

Reply:
Exactly.

Reply:
It allows it, but it picks the wrong function declaration unless an array is forced as the last parameter, negating it's usefulness.

Reply:
A bit of clarification... the reason matches isn't the last parameter is mostly for readability purposes, since the `--foo` is the most important part of this class, I want it easy to find and read.  The parameter ambiguity can be circumvented using dedicated `null` placeholders in the constructors.

Reply:
Done via 8d242b7.